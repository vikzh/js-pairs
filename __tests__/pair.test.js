import {
  cons, car, cdr, toString,
} from '../src';

describe('Pair', () => {
  it('#cons', () => {
    const pair = cons(1, 2);
    expect(car(pair)).toBe(1);
    expect(cdr(pair)).toBe(2);
  });

  it('#toString', () => {
    const pair = cons('text1', 'text2');
    expect(toString(pair)).toBe('(text1, text2)');
  });

  it('#toString2', () => {
    const pair = cons(5, cons(7, 9));
    expect(toString(pair)).toBe('(5, (7, 9))');
  });

  it('#checkPair', () => {
    expect(() => {
      car('test');
    }).toThrowErrorMatchingSnapshot();

    expect(() => {
      cdr(2345);
    }).toThrowErrorMatchingSnapshot();

    expect(() => {
      car(() => 1);
    }).toThrowErrorMatchingSnapshot();

    expect(() => {
      car({ key: 'testValue' });
    }).toThrowErrorMatchingSnapshot();
  });
});
