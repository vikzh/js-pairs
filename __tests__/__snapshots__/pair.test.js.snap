// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pair #checkPair 1`] = `"Argument must be pair, but it was 'test'"`;

exports[`Pair #checkPair 2`] = `"Argument must be pair, but it was '2345'"`;

exports[`Pair #checkPair 3`] = `"Argument must be pair, but it was '() => 1'"`;

exports[`Pair #checkPair 4`] = `
"Argument must be pair, but it was '{
  \\"key\\": \\"testValue\\"
}'"
`;
